common/src/chain/transaction/output.rs

```diff
@@ -41,26 +41,33 @@ pub enum OutputPurpose {
     #[codec(index = 1)]
     LockThenTransfer(Destination, OutputTimeLock),
     #[codec(index = 2)]
-    StakePool(Box<StakePoolData>),
-    #[codec(index = 3)]
     Burn,
+    /// Output type that is used to create a stake pool
+    #[codec(index = 3)]
+    StakePool(Box<StakePoolData>),
+    /// Output type that represents spending of a stake pool output in a block reward
+    /// in order to produce a block
+    #[codec(index = 4)]
+    SpendStakePool(Box<StakePoolData>),
```

Comment:
If you're gonna keep `StakePoolData` as the contents of the arm, what's exactly the point of creating a new arm?

Reply:
good question. Added the type to separate logic of processing the outputs. But now agree that it seems like possible with only one type

Reply:
Like I mentioned before, the benefit of using a different type is saving space in the blockchain by having only the minimal relevant information to stake.

Reply:
I see. I could use something like `ProduceBlockFromStake(OutPoint)`. Thus it will reference to the stake data and save space on every block. I'll give it a try, do you agree?

Reply:
I don't think it's enough. You need a little more information, like the stake pool id, at least. Maybe you can strangle that from the input? I don't know. Give it a shot.

Reply:
I mean using OutPoint I can read StakePool output and get all the information from it

Reply:
Apparently having only `OutPoint` is not enough to retrieve information to verify transactions in `TransactionVerifier` because the transaction index is optional. So I have to put at least some amount of information into the output. 

I'd propose not to proceed with this at the moment and just clone `StakePoolData` because I'm not sure which data fields are going to be used to implement all the operations we want. I think it's better to place a TODO and optimize it later.

Reply:
I don't mind, but I don't understand the logic. You could copy the struct and then remove those you don't need.